src/stream/src/executor/eowc/eowc_gap_fill.rs

```
@@ -71,76 +70,84 @@ struct ExecutionVars<S: StateStore> {
 }
 
 impl<S: StateStore> ExecutorInner<S> {
-    fn generate_filled_rows(
-        prev_row: &OwnedRow,
-        curr_row: &OwnedRow,
+    #[try_stream(ok = StreamChunk, error = ExprError)]
+    async fn generate_filled_rows<'a>(
+        prev_row: &'a OwnedRow,
+        curr_row: &'a OwnedRow,
         time_column_index: usize,
-        fill_columns: &HashMap<usize, FillStrategy>,
+        fill_columns: &'a HashMap<usize, FillStrategy>,
         interval: risingwave_common::types::Interval,
-        metrics: &GapFillMetrics,
-    ) -> Result<Vec<OwnedRow>, ExprError> {
-        let mut filled_rows = Vec::new();
+        metrics: &'a GapFillMetrics,
+        chunk_builder: &'a mut StreamChunkBuilder,
+    ) {
         let (Some(prev_time_scalar), Some(curr_time_scalar)) = (
             prev_row.datum_at(time_column_index),
             curr_row.datum_at(time_column_index),
         ) else {
-            return Ok(filled_rows);
+            return Err(ExprError::InvalidParam {
+                name: "time_column",
+                reason: "Time column contains NULL values".into(),
```

Is it possible for the user to somehow provide null values in their `prev_row` and `curr_row`? If so we should just return an error log and just not gapfill.

---

I suppose returning `Ok(())` is fine here.

---

> Is it possible for the user to somehow provide null values in their `prev_row` and `curr_row`? If so we should just return an error log and just not gapfill. Otherwise we might just have crash loop which is poor UX.

Yes, it is possible for now. But I've tested this scenario in Tiger Data, and they return an error like `ERROR:  invalid time_bucket_gapfill argument: ts cannot be NULL`.

---

I think I haven't handled null time-column values properly as well. Maybe it's better to disallow gap fill when the time column contains null values.
I notice that in over_window we panic with `.expect("order key column must be non-NULL")`. Should I panic or return an error when gap fill encounters a null time column? @kwannoel

---

We can follow the behaviour of over_window and panic.